In [0]:
import pandas as pd
import scipy.stats as stats

In [4]:
url="http://scipy-lectures.org/_downloads/brain_size.csv"
data = pd.read_csv(url, sep=';', na_values='.', index_col=0)
data.head()

Gender  FSIQ  VIQ  PIQ  Weight  Height  MRI_Count
1  Female   133  132  124   118.0    64.5     816932
2    Male   140  150  124     NaN    72.5    1001121
3    Male   139  123  150   143.0    73.3    1038437
4    Male   133  129  128   172.0    68.8     965353
5  Female   137  132  134   147.0    65.0     951545

In [5]:
data.describe()

FSIQ         VIQ        PIQ      Weight     Height     MRI_Count
count   40.000000   40.000000   40.00000   38.000000  39.000000  4.000000e+01
mean   113.450000  112.350000  111.02500  151.052632  68.525641  9.087550e+05
std     24.082071   23.616107   22.47105   23.478509   3.994649  7.228205e+04
min     77.000000   71.000000   72.00000  106.000000  62.000000  7.906190e+05
25%     89.750000   90.000000   88.25000  135.250000  66.000000  8.559185e+05
50%    116.500000  113.000000  115.00000  146.500000  68.000000  9.053990e+05
75%    135.500000  129.750000  128.00000  172.000000  70.500000  9.500780e+05
max    144.000000  150.000000  150.00000  192.000000  77.000000  1.079549e+06

#1 sample T test

In this example, after seeing the stats, it seems this sample's avg IQ is 113, 112 and 111 are different from the standard avg IQ which is 100. Let's test this hypothesis.

Null Hypothesis: Average IQ of the population is 100.
Alternate Hypo: Avg IQ is not 100.

Let significance level(alpha) is 5%

In [8]:
IQs = ['FSIQ', 'VIQ', 'PIQ']
for IQ in IQs:
  print(stats.ttest_1samp(data[IQ], 100))

Ttest_1sampResult(statistic=3.532307014238269, pvalue=0.0010766792736967715)
Ttest_1sampResult(statistic=3.3074146385401786, pvalue=0.002030117404781822)
Ttest_1sampResult(statistic=3.1030246997178783, pvalue=0.0035555593418294417)


It turns out that p value is <0.05 for all IQs and we can confidently reject null hypothesis. It leads to the insight that the population from which this sample is drawn will be biased towards high IQ people rather than just normal people who have avg IQ 100.

**Inference about the population:**
AvgIQ of the population from which this sample taken is not 100.

# 2 sample T test

Let's conduct a 2 sample test on male vs female

In [17]:
grouped = data.groupby('Gender')
for IQ in IQs:
  for gender, value in grouped[IQ]:
    print(gender, value.mean())

Female 111.9
Male 115.0
Female 109.45
Male 115.25
Female 110.45
Male 111.6


It seems that the avg IQs of men is more than women which is not significant in reality. Let's test this hypothesis. 
H0: The difference b/w IQs of male and female is not significant
Ha: The difference is significant.

In [18]:
male = data[data['Gender'] == 'Male']
female = data[data['Gender'] == 'Female']
male.head()

Gender  FSIQ  VIQ  PIQ  Weight  Height  MRI_Count
2    Male   140  150  124     NaN    72.5    1001121
3    Male   139  123  150   143.0    73.3    1038437
4    Male   133  129  128   172.0    68.8     965353
9    Male    89   93   84   134.0    66.3     904858
10   Male   133  114  147   172.0    68.8     955466

In [19]:
for IQ in IQs:
  print(stats.ttest_ind(male[IQ], female[IQ]))

Ttest_indResult(statistic=0.4026724743703011, pvalue=0.6894456253897778)
Ttest_indResult(statistic=0.7726161723275011, pvalue=0.44452876778583217)
Ttest_indResult(statistic=0.15980113150762698, pvalue=0.8738841403250049)


It turns out that the p value is not < 0.05 for any case. We failed to reject null hypothesis here. In other words, we failed to claim mean IQ of both male and female is statistically significant.